pkg/action/pull.go

```diff
@@ -138,6 +140,36 @@ func (p *Pull) Run(chartRef string) (string, error) {
 		return out.String(), err
 	}
 
+	// Print a pull summary for repo mode and non-OCI downloads.
+	// Some repos (e.g., Bitnami) resolve --repo pulls to oci:// refs, but that
+	// path does not go through the registry client's summary printer, so we emit
+	// a consistent summary here. We mirror the OCI output format:
+	//
+	//   Pulled: <chartUrl>:<version>
+	//   Checksum: sha256:<digest>
+	//
+	// For HTTP/repo pulls, the digest is the SHA-256 of the downloaded .tgz archive.
+	// For direct OCI pulls, the registry client already prints "Pulled:" and
+	// "Digest:" (manifest digest), so we do not print here to avoid duplicates.
+	if p.RepoURL != "" || !registry.IsOCI(downloadSourceRef) {
+		base := strings.TrimSuffix(downloadSourceRef, ".tgz")
+		chart, ver := splitChartNameVersion(base)
+
+		tag := chart
+		if ver != "" {
+			tag += ":" + ver
+		} else if p.Version != "" {
+			tag += ":" + p.Version
+		}
+		fmt.Fprintf(&out, "Pulled: %s\n", tag)
+
+		if sum, err := sha256File(saved); err == nil {
+			fmt.Fprintf(&out, "Checksum: sha256:%x\n", sum)
+		} else {
+			fmt.Fprintf(&out, "Checksum failed: %v\n", err)
+		}
```

Comment:
Internally, `DownloadTo` will sometimes have the hash of a file. The digest/hash is one of the things on a Repo index. I wonder if that should be exposed and checked against the calculated hash of the file. Then a message displayed if they are different.

Thoughts?

Reply:
Ok.. I will see how I can add a test for that but here is the output with a manually corrupted file.

```sh
$ HELM_TEST_CORRUPT=1  bin/helm pull stable/mysql

Pulled: stable/mysql
Checksum: sha256:3791aa11aed2d587c575be44fdd56895d0ecf62a84fe3a326e51ef0fc03208be
WARNING: Checksum mismatch!
  Expected: 35f232f0b4df50e85c6d65cbe7e5291b8ac8fab2c5a6afa8a0b816c7ce594390
  Got:      3791aa11aed2d587c575be44fdd56895d0ecf62a84fe3a326e51ef0fc03208be
```